src/input/reader_to_storage.cc

```diff
@@ -347,17 +347,73 @@ StorageBase * ReaderToStorage::make_storage(PathBase &p, const Type::Array *arra
                     << EI_ErrorAddress(p.as_string()) << EI_InputType(array->desc()) );
         }
     } else {
-        // try automatic conversion to array with one element
-        if ( array->match_size( 1 ) ) {
-            StorageArray *storage_array = new StorageArray(1);
-            const Type::TypeBase &sub_type = array->get_sub_type();
-            storage_array->new_item(0, make_storage(p, &sub_type) );
-
-            return storage_array;
-        } else {
-            THROW( ExcInputError() << EI_Specification("Automatic conversion to array not allowed. The value should be '" + p.get_node_type(ValueTypes::array_type) + "', but we found: ")
-                    << EI_ErrorAddress(p.as_string()) << EI_JSON_Type( p.get_node_type(p.get_node_type_index()) ) << EI_InputType(array->desc()) );
+    	// if transposition is carried, only conversion to array with one element is allowed
+    	if (try_transpose_read_) {
+			// try automatic conversion to array with one element
+			if ( array->match_size( 1 ) ) {
+				StorageArray *storage_array = new StorageArray(1);
+				const Type::TypeBase &sub_type = array->get_sub_type();
+				storage_array->new_item(0, make_storage(p, &sub_type) );
+
+				return storage_array;
+			} else {
+				THROW( ExcInputError() << EI_Specification("Automatic conversion to array not allowed. The value should be '" + p.get_node_type(ValueTypes::array_type) + "', but we found: ")
```

Comment:
Adapt the message to the new context, e.g.
During transpose auto-conversion, the conversion to the single element array not allowed. Require type:
... \nFound on input: ...\n